solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/outbound/JCSMPOutboundMessageHandler.java

```diff
@@ -96,52 +101,106 @@ public void handleMessage(Message<?> message) throws MessagingException {
 					String.format("Unable to parse header %s", SolaceBinderHeaders.CONFIRM_CORRELATION), e);
 		}
 
-		XMLMessage xmlMessage = xmlMessageMapper.map(message, properties.getExtension().getHeaderExclusions(),
-				properties.getExtension().isNonserializableHeaderConvertToString());
-		correlationKey.setRawMessage(xmlMessage);
-		xmlMessage.setCorrelationKey(correlationKey);
+		List<XMLMessage> smfMessages;
+		List<Destination> dynamicDestinations;
+		if (message.getHeaders().containsKey(SolaceBinderHeaders.BATCHED_HEADERS)) {
+			LOGGER.debug("Detected header {}, handling as batched message (Message<List<?>>) <message handler ID: {}>",
+					SolaceBinderHeaders.BATCHED_HEADERS, id);
+			smfMessages = xmlMessageMapper.mapBatchMessage(
+					message,
+					properties.getExtension().getHeaderExclusions(),
+					properties.getExtension().isNonserializableHeaderConvertToString());
+
+			BatchProxyCorrelationKey batchProxyCorrelationKey = transactedSession == null ?
+					new BatchProxyCorrelationKey(correlationKey, smfMessages.size()) : null;
+			smfMessages.forEach(smfMessage -> smfMessage.setCorrelationKey(
+					Objects.requireNonNullElse(batchProxyCorrelationKey, correlationKey)));
+
+			// after successfully running xmlMessageMapper.mapBatchMessage(),
+			// SolaceBinderHeaders.BATCHED_HEADERS is verified to be well-formed
 
-		LOGGER.debug("Publishing message to destination [ {}:{} ]",
-				targetDestination instanceof Topic ? "TOPIC" : "QUEUE", targetDestination);
+			@SuppressWarnings("unchecked")
+			List<Map<String, Object>> batchedHeaders = (List<Map<String, Object>>) message.getHeaders()
+					.getOrDefault(SolaceBinderHeaders.BATCHED_HEADERS,
+							Collections.nCopies(smfMessages.size(), Collections.emptyMap()));
+
+			dynamicDestinations = batchedHeaders.stream()
+					.map(h -> getDynamicDestination(h, correlationKey))
+					.toList();
+		} else {
+			XMLMessage smfMessage = xmlMessageMapper.map(
+					message,
+					properties.getExtension().getHeaderExclusions(),
+					properties.getExtension().isNonserializableHeaderConvertToString());
+			smfMessage.setCorrelationKey(correlationKey);
+			smfMessages = List.of(smfMessage);
+			dynamicDestinations = Collections.singletonList(getDynamicDestination(message.getHeaders(), correlationKey));
+		}
+
+		correlationKey.setRawMessages(smfMessages);
 
 		try {
-			producer.send(xmlMessage, targetDestination);
+			for (int i = 0; i < smfMessages.size(); i++) {
+				XMLMessage smfMessage = smfMessages.get(i);
+				Destination targetDestination = Objects.requireNonNullElse(dynamicDestinations.get(i), configDestination);
+
+				LOGGER.debug("Publishing message {} of {} to destination [ {}:{} ] <message handler ID: {}>",
+						i + 1, smfMessages.size(), targetDestination instanceof Topic ? "TOPIC" : "QUEUE",
+						targetDestination, id);
+
+				producer.send(smfMessage, targetDestination);
```

Comment:
I wonder if [Solace's sendMultiple()](https://docs.solace.com/API-Developer-Online-Ref-Documentation/java/com/solacesystems/jcsmp/XMLMessageProducer.html#sendMultiple-com.solacesystems.jcsmp.JCSMPSendMultipleEntry:A-int-int-int-) could be of any help? 
@mgevantmakher Do you have experience with sendMultiple()? Any thoughts?